### PR TITLE
Add data validation spec

### PR DESCRIPTION
**Pull request by emilyliu7321**
PR content from @bnewbold

### CODE REVIEW

**Comment by bnewbold** [comment]
@devinivy added you for a set of eyes on this. Would be nice to have out at same time as other docs push, but not particularly urgent.

I might also move this content to the existing data model page instead of a new page, but I can hit that after review.

File: content/specs/data-validation.md [hunk 1]
```diff
@@ -0,0 +1,52 @@
+---
+title: Data Validation
+summary: Expectations around recommended data limits and validation
+---
+
+# Data Validation
+
+Some software does not need to worry much about Bluesky or atproto schema validation. For example, client apps and bots can generally assume that data they receive from the Bluesky API is valid, and when they try to create records the PDS is responsible for double-checking schema validation.
+
+Software which consumes directly from the event stream (firehose) should be more careful. And developers designing new tools or clever hacks should be away of some expectations and hard limits around data validation.
```
Comment (surfdude29) [inline content/specs/data-validation.md:10]: A typo here I think, 'aware'?

File: content/specs/data-validation.md [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+
+If an individual record fails to validate for any reason, the entire record should be ignored, but other records from the same repository should be processed. If there is a problem with the repository commit data or repository structure ("MST"), the repository should be market as invalid, but old content does not need to be de-indexed or purged. As soon as a new valid commit is received, the repository should be marked valid again. This logic mostly relevant to feed generators, AppViews, and moderation services, all of which may be consuming from the firehose.
```
Comment (surfdude29) [inline content/specs/data-validation.md:12]: 'should be market as invalid' > 'should be marked as invalid'

Comment (bnewbold) [reply content/specs/data-validation.md]: fixed, thanks!

File: content/specs/data-validation.md (hunk 1 above)
Comment (bnewbold) [reply content/specs/data-validation.md]: fixed, thanks!

**Comment by bnewbold** [comment]
@dholms might actually be in a better position to review this.

It was originally written to be posted in a different context; I can handle re-phrasing things to make more sense here in the atproto specs. Mostly looking for a sanity check on the values and validation process.

**Comment by bnewbold** [comment]
Closing in lieu of https://github.com/bluesky-social/atproto-website/pull/378

**Pull request closed without merging**